### ISSUE TITLE
Saved Homebrews not loading

### ISSUE THREAD

**Issue by tpcantwell** [stolksdorf/naturalcrit#95]
Google Chrome Version 50.0.2661.102 m
Windows 7

Version 2 looks incredible, but saved homebrews do not seem to be loading. Here is the specific homebrew I am trying to access
http://www.naturalcrit.com/homebrew/edit/B1eK3DRvM

**Comment by stolksdorf** [comment]
Really weird bug, I'll take a deeper look into it tonight.

**Comment by tpcantwell** [comment]
Hmmm interesting the image is stored in google photos just for your reference. For future use if working from Google Photos causes issues do you have a suggestions for where to host images that I want to use in my homebrews? I was going to use imgur, but the png file I wanted to store was too large and it kept converting to jpeg losing its alpha layer. Thats why I went to Google photos to store the png

**Comment by stolksdorf** [comment]
Google photos should be fine, just double check your permissions on the uploaded image (Try opening the image url in Incognito mode, if you can view it there, Homebrewery can load it). 

What concerns me is that a 403'd image shouldn't break the page like that. It should just not show the image. I'll keep you updated with the fix.

**Comment by tpcantwell** [comment]
Ok thanks. Also I opened the image in icognito mode and it opened fine

**Comment by stolksdorf** [comment]
Turns out it has nothing to do with the image. You have two unclosed style tags at the end of your brew. This is invalid HTML and _shouldn't_ have worked before, but HTML is pretty tolerate. 

So why did it break now? In v2 I started doing server-side rendering to speed up page loads. When the page is rendered, these opened tags tell the browser that the rest of the page after them should be treated as styles and not executed. This includes the editor and all the executed javascript on the page. 

So how can we fix it. Easiest solution is just copy your brew from [here](http://www.naturalcrit.com/homebrew/source/BkKhw0Df), paste it into a new brew and remove the two style tags at the bottom and re-save it.

**Comment by stolksdorf** [comment]
**tl;dr:** It's fixed, you don't need to make a new brew or anything.

After thinking about the issue more, since the target audience of this tool probably doesn't have too much HTML knowledge, being hit with this bug could be pretty annoying, since it only happens on a fresh reload of the page (since it's re-rendered on the server first).

On-the-fly HTML validation and informing the user of their errors and what not is not that simple to implement as well. Also any exsisting brews with malformed html in them already have their share pages broken. 

All-in-all a bad problem. The soltuion is just deployed is that if The Homebrewery is rendering server side it doesn't try and render any of the brew pages, just dummy pages (used in the partial brew rendering process). As soon as it loads client side, it'll render the brew normally.

**Comment by tpcantwell** [comment]
Hey thanks for the help. Also side note. I am really glad you started the Patreon page! If people know they can help I know they will be willing to support this tool at least to help keep the servers running. Keep it up
